Add image inpainting network (#865)
opencv/open_model_zoo: models/public/gmcnn-tf/gmcnn-tf.md

> # gmcnn-tf

Reviewer: I think it would be prudent to call this `gmcnn-places2-tf`, since there are several other GMCNN models in the linked repository.

## Use Case and High-Level Description

The `gmcnn-tf` is the TensorFlow implementation of GMCNN Image Inpainting model,

> aimed to estimate suitable pixel information tofill holes in images. The `gmcnn-tf`

Reviewer: ```suggestion
aimed to estimate suitable pixel information to fill holes in images. `gmcnn-tf`
```

Reviewer: Don't forget to update the config, as well.

is trained on Places2 dataset with free-form masks. Originally redistributed as checkpoint files,

> was converted to frozen graph.For details see repository <https://github.com/shepnerd/inpainting_gmcnn>.

Reviewer: ```suggestion
it was converted to a frozen graph. For details see [repository](https://github.com/shepnerd/inpainting_gmcnn).
```

### Steps to Reproduce Conversion to Frozen Graph

1. Clone the original repository
```sh
git clone https://github.com/shepnerd/inpainting_gmcnn.git
cd inpainting_gmcnn/tensorflow
```
2. Checkout the commit that the conversion was tested on:
```sh
git checkout ba7f710
```
3. Install the [original dependencies](https://github.com/shepnerd/inpainting_gmcnn#prerequisites).

> (TensoFlow\* version used - 1.14.0, CPU).

Reviewer: ```suggestion
(TensorFlow\* version used - 1.14.0, CPU).
```

4. Download the [pretrained weights](https://drive.google.com/file/d/1aakVS0CPML_Qg-PuXGE1Xaql96hNEKOU/view?usp=sharing)

> 5. Run sample [conversion script](./freeze_model.py):

Reviewer: Something seems to be missing here. If you run the script as written, it's not going to work, since `sys.path` does not, by default, include the current directory.

  where:

    - B - batch size
    - H - image height
    - W - image width
    - C - number of channels

   Expected color order: RGB.

2. Mask, name: `Placeholder_1`, shape: [1x512x680x1], format: [BxHxWxC]
  where:

    - B - batch size
    - H - mask height
    - W - mask width
    - C - number of channels

### Converted Model

1. Image, name: `Placeholder`, shape: [1x3x512x680], format: [BxCxHxW]
  where:

    - B - batch size
    - C - number of channels
    - H - image height
    - W - image width

   Expected color order: BGR.

2. Mask, name: `Placeholder_1`, shape: [1x1x512x680], format: [BxCxHxW]
  where:

    - B - batch size
    - C - number of channels
    - H - mask height
    - W - mask width

## Output

### Original Model

Restored image, name `Cast`, shape: [1x512x680x3], format: [BxHxWxC]
  where:

    - B - batch size
    - H - image height
    - W - image width
    - C - number of channels

   Expected color order: RGB.

### Converted Model

Restored image, name: `Cast`, shape: [1x3x512x680], format: [BxCxHxW]
  where:

    - B - batch size
    - C - number of channels
    - H - image height
    - W - image width

>    Expected color order: BGR.

Reviewer: Something isn't right here. In the documentation, the color order is flipped in the converted model, but there's no `--reverse_input_channels` flag in the config file.

Author: There was a bug in documentation, fixed
